HABApp/openhab/http_connection.py

```diff
@@ -325,22 +333,16 @@ async def async_thing_remove_link(self, thing_uid: str, channel: str, item_name:
         if self.config.general.listen_only:
             return False
 
-        # rest/links/ endpoint needs the channel to be url encoded 
-        # (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
-        # otherwise the REST-api returns HTTP-Status 500 InternalServerError
-        channel_uid = urllib.parse.quote(f"{thing_uid}:{channel}")
+        channel_uid = f"{thing_uid}:{channel}"
         url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}', item=item_name, channel=channel_uid)
```

Comment:
```suggestion
        url = self.__get_openhab_url('rest/links/{item:s}/{thing_uid:s}:{channel:s}', item=item_name, thing_uid=thing_uid, channel=channel_uid)
```

Reply:
I added a self.__get_link_url() method to have it in one place. This also gets rid of the check in __get_openhab_url.